### ISSUE TITLE
Inconsistencies between source code and code in examples

### ISSUE THREAD

**Issue by thisiswhereitype** [powerjg/learning_gem5#18]
In Chapter 1:
The cmd at the end of http://learning.gem5.org/book/part1/cache_config.html:
 `build/X86/gem5.opt configs/tutorial/two_level_opts.py --l2_size='1MB' --l1d_size='128kB'`
However in the head of the gem5 repository you have the sample code already from Chapter 1, under `configs/learning_gem5/part1` which contradicts `configs/tutorial/` used in the text. There does not appear to be any reference to the first path in the book.

In Chapter 2:
The cmd taken from the gem5 ouput at http://learning.gem5.org/book/part2/helloobject.html:
`build/X86/gem5.opt configs/learning_gem5/part2/run_hello.py`
This now refers to `configs/learning_gem5/part2/` dir again. It is also confusing as there are now two `learning_gem5` directories in `configs` and `src`.

Finally it would be very helpful if the sections in general provided a summary of which files are placed where for the examples and any commands run. Perhaps a shell script can unpack them, given a gem5 repo path? It will be much easier for the user to reproduce the examples.

I am willing to make these changes however I cannot guarantee the time soon. Thanks for producing such a helpful book so far!

**Comment by powerjg** [comment]
Thanks for the feedback! I'll hopefully find some time to go through the backlog of issues soon. This one will be near the top of the list.

**Comment by powerjg** [comment]
Hi Joe,

I'm looking at this now as I'm editing the book for the upcoming ASPLOS tutorial.

I'm not sure what to do about this. The code for the complete file after the parameters chapter (http://learning.gem5.org/book/part2/parameters.html) is contained in the gem5 source tree now (src/learning_gem5). However, the point of this section is for the reader to create their own directory and put the code there. I'm not sure the best way to show examples. I'm open to suggestions.

Thanks again for the feedback.

**Comment by thisiswhereitype** [comment]
Apologies for my slow response,

Rereading the original post I now have no issue understanding how this code fits together having seen it in the source tree. I looked again at the chapters linked - my experience is that I found it disorientating jumping between code snippets when there are multiple files although is some cases the book is explicit.
I suggest a tree diagram showing paths to files and a brief comment of their significance in the overall structure. Much like the `tree`. My thought is the chapters are trying to communicate the idea and purpose of these files. They and their comments should be read for the full implementation.

Thanks for returning to this issue,
I hope you find my suggestion relevant and helpful.
